invoices/invoice_expiry_watcher.go

```diff
@@ -190,7 +192,13 @@ func (ew *InvoiceExpiryWatcher) cancelNextExpiredInvoice() {
 			return
 		}
 
-		err := ew.cancelInvoice(top.PaymentHash)
+		// Don't force-cancel already accepted invoices. An exception to
+		// this are auto-generated keysend invoices. Because those start
+		// out in the Accepted state, the expiry field would never be
```

Comment:
the invoices don't technically start out in the accepted state tho, right? they still do transition from `ContractOpen` to `ContractAccepted`?

Reply:
Yes, the go through the normal lifecycle.

Reply:
Comment updated